File: python/sorting/bubble_sort.py
from .sort import SortType, Sort


class BubbleSort(Sort):
    def sort(self, arr, type=SortType.asc):
        length = len(arr)

        for i in range(length):
            for j in range(length - i - 1):
                if self.compare(arr[j], arr[j + 1], type):
                    arr[j], arr[j + 1] = arr[j + 1], arr[j]

        return arr

File: scala/src/main/scala/sorting/BubbleSort.scala
package sorting

class BubbleSort() extends Sort {
  def sort(arr: Array[Int], sortType: SortType): Array[Int] = {
    val length: Int = arr.length

    for (i <- Range(0, length, 1))
      for (j <- Range(0, length - i - 1, 1))
        if (compare(arr(j), arr(j + 1), sortType)) {
          swap(arr, j, j + 1)
        }
    arr
  }

}
